packages/Tabs/Tabs.jsx

```diff
@@ -16,42 +23,101 @@ const Tabs = props => {
   const tabsRoot = useRef()
   const [tabsContainerWidth, setTabsContainerWidth] = useState()
   const [tabsTranslatePosition, setTabsTranslatePosition] = useState(0)
+  const [totalTabsWidth, setTotalTabsWidth] = useState(0)
+  const [firstTabWidth, setFirstTabWidth] = useState(0)
   const [isScrollEnabled, setScrollEnabled] = useState(false)
-  const [isLeftArrow, setIsLeftArrow] = useState(false)
-  const [isRightArrow, setIsRightArrow] = useState(false)
+  const [isLeftArrowVisible, setLeftArrowVisible] = useState(false)
+  const [isRightArrowVisible, setRightArrowVisible] = useState(false)
   const [current, setCurrent] = useState(0)
-  const fullWithRef = useRef(0)
-  const { children, ...rest } = props
+  const tabRef = useRef(null)
+  const { children, leftArrowLabel, rightArrowLabel, ...rest } = props
 
-  const currentIndex = i => {
-    setCurrent(i)
-  }
-  const _nodes = new Map()
   const getTabsWidth = () => {
-    let totalTabsWidth = 0
-    Object.keys(_nodes).forEach(key => {
-      const value = _nodes[key]
-      if (value && value.node && value.node.offsetWidth) {
-        totalTabsWidth += value.node.offsetWidth
+    let tabsWidthValue = 0
+    const marginBuffer = 28
+    const tabsArray =
+      tabRef.current && tabRef.current.children[0] && tabRef.current.children[0].childNodes
+    const firstTab =
+      tabRef.current && tabRef.current.children[0] && tabRef.current.children[0].firstChild
+    tabsArray.forEach(value => {
+      if (value && value.offsetWidth) {
+        tabsWidthValue += value.offsetWidth + marginBuffer
       }
     })
-
+    const firstTabValue = firstTab.offsetWidth - marginBuffer * 3
```

Comment:
maybe we can create a const for 'marginBuffer * 3'.  Also not sure in TDS Community, but maybe we can declare these consts at the top with all caps?